src/harness/testcases/WINNF_FT_S_SIQ_testcase.py

```diff
@@ -206,6 +206,154 @@ def test_WINNF_FT_S_SIQ_8(self):
+        ppa_record1, [pal_record1], pal_low_frequency1, pal_high_frequency1,
+        device_a['userId'])
+
+    # Move device_a into the first PPA zone
+    device_a['installationParam']['latitude'], device_a['installationParam'][
+        'longitude'] = getRandomLatLongInPolygon(ppa_record1)
+
+    # Second PPA with device_c and FR2 = 3600 - 3610
+    pal_low_frequency2 = 3600000000
+    pal_high_frequency2 = 3610000000
+    pal_record2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'pal_record_1.json')))
+    ppa_record2 = json.load(
+        open(os.path.join('testcases', 'testdata', 'ppa_record_1.json')))
+    ppa_record2, pal_record2 = makePpaAndPalRecordsConsistent(
+        ppa_record2, [pal_record2], pal_low_frequency2, pal_high_frequency2,
+        device_c['userId'])
+
+    # Move device_c into the second PPA zone
+    device_c['installationParam']['latitude'], device_c['installationParam'][
+        'longitude'] = getRandomLatLongInPolygon(ppa_record2)
+
+    # Register 4 devices.
+    cbsd_ids = self.assertRegistered([device_a, device_c, device_e, device_f])
+
+    # Update PPA record with device_a's CBSD ID and Inject data
+    ppa_record1['ppaInfo']['cbsdReferenceId'] = [cbsd_ids[0]]
+    self._sas_admin.InjectPalDatabaseRecord(pal_record1[0])
+    zone_id = self._sas_admin.InjectZoneData({'record': ppa_record1})
+    self.assertTrue(zone_id)
+
+    # Update PPA record with device_c's CBSD ID and Inject data
+    ppa_record2['ppaInfo']['cbsdReferenceId'] = [cbsd_ids[1]]
+    self._sas_admin.InjectPalDatabaseRecord(pal_record2[0])
+    zone_id = self._sas_admin.InjectZoneData({'record': ppa_record2})
+    self.assertTrue(zone_id)
+
+    # Create Spectrum Inquiry requests
+    spectrum_inquiry_0 = {
+        'cbsdId': cbsd_ids[0],
+        'inquiredSpectrum': [{
+            'lowFrequency': 3550000000,
+            'highFrequency': 3700000000
+        }]
+    }
+    spectrum_inquiry_1 = {
+        'cbsdId': cbsd_ids[1],
+        'inquiredSpectrum': [{
+            'lowFrequency': 3600000000,
+            'highFrequency': 3700000000
+        }]
+    }
+    spectrum_inquiry_2 = {
+        'cbsdId': cbsd_ids[2],
+        'inquiredSpectrum': [{
+            'lowFrequency': 3550000000,
+            'highFrequency': 3700000000
+        }]
+    }
+    spectrum_inquiry_3 = {
+        'cbsdId': cbsd_ids[3],
+        'inquiredSpectrum': [{
+            'lowFrequency': 3550000000,
+            'highFrequency': 3700000000
+        }]
+    }
+    request = {
+        'spectrumInquiryRequest': [
+            spectrum_inquiry_0, spectrum_inquiry_1, spectrum_inquiry_2,
+            spectrum_inquiry_3
+        ]
+    }
+
+    # Check Spectrum Inquiry response
+    response = self._sas.SpectrumInquiry(request)['spectrumInquiryResponse']
+    for resp_num, resp in enumerate(response):
+      self.assertEqual(resp['cbsdId'], cbsd_ids[resp_num])
+      self.assertEqual(resp['response']['responseCode'], 0)
+
```

Comment:
I am not sure that this additional passing by all the responses for checking those two conditions is better than adding them in the later loops.
do you agree?

Reply:
Agreed. Removed this and added them to the next loop.